dotcom-rendering/src/lib/useCommentCount.ts

```diff
@@ -4,32 +4,35 @@ import { useApi } from './useApi';
 
 type CommentCounts = Record<string, number>;
 
-const getInitialSetFromDOMAttribute = (attribute: string) =>
-	[...document.querySelectorAll(`[${attribute}]`)]
-		.map((element) => element.getAttribute(attribute))
-		.filter(isNonNullable);
+const DATA_ATTRIBUTE = 'data-discussion-id';
 
-const uniqueDiscussionIds = new Set<string>(
-	isServer ? [] : getInitialSetFromDOMAttribute('data-discussion-id'),
-);
+const uniqueDiscussionIds = isServer
+	? undefined
+	: new Set<string>(
+			// create an initial set of IDs by reading what is in the DOM
+			[...document.querySelectorAll(`[${DATA_ATTRIBUTE}]`)]
+				.map((element) => element.getAttribute(DATA_ATTRIBUTE))
+				.filter(isNonNullable),
+	  );
+
+const getUrl = (base: string) =>
+	uniqueDiscussionIds
+		? `${base}/getCommentCounts?${new URLSearchParams({
+				'short-urls': [...uniqueDiscussionIds]
+					.sort() // ensures identical sets produce the same query parameter
+					.join(','),
+		  }).toString()}`
+		: undefined;
 
 export const useCommentCount = (
 	discussionApiUrl: string,
 	shortUrl: string,
 ): number | undefined => {
-	if (!isServer) {
-		uniqueDiscussionIds.add(shortUrl);
-	}
+	uniqueDiscussionIds?.add(shortUrl);
 
-	const searchParams = new URLSearchParams({
-		'short-urls': [...uniqueDiscussionIds]
-			.sort() // ensures identical sets produce the same query parameter
-			.join(','),
-	});
-	const url = `${discussionApiUrl}/getCommentCounts?${searchParams.toString()}`;
-	const { data } = useApi<CommentCounts>(url, {
+	const { data } = useApi<CommentCounts>(getUrl(discussionApiUrl), {
```

Comment:
I think it's clear, but would leaving a note saying that `uniqueDiscussionIds` only gets generated on the client, and thus this only runs on the client, outlining the issue we've just had might avoid a good citizen trying to simplify later falling into the same trap?

I found [this link](https://swr.vercel.app/docs/conditional-fetching) useful too as it's a bit of a magic API. `null | undefined` means ignore.

Reply:
Not sure if just inlining this method would be easier to read, but would mean we'd have to align [`useApi`'s `url` property](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/lib/useApi.tsx#L34) to resemble [SWR's `Key` type](https://github.com/vercel/swr/blob/main/_internal/src/types.ts#L306)

Reply:
I’ve updated the code to add more documentation and make the method more decriptive: `maybeGetUrl`.